---
title: "GWAS_GAPIT_BLINK"
author: "Leah Treffer"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Objective

Run a GWAS analysis for measured traits 
	C7 Salina 2018, 2019, 2020
	C10 Salina 2021, 2022
	C10 Olathe 2021, 2022
Compare results to those from 1.1_GWAS_FarmCPU

```{r}
rm(list=ls(all=TRUE))
#set current working directory
#setwd()
setwd('/Volumes/Backup_Plus/AM/New_GWAS/2023.01.27/IWG_GWAS')
```

```{r get_functions, include=FALSE}
# GAPIT - Genomic Association and Prediction Integrated Tool
# Designed by Zhiwu Zhang
# Written by Zhiwu Zhang, Alex Lipka, Feng Tian, You Tang and Jiabo Wang
# Last update: Nov 15, 2021 for version 3

#Install packages (Do this section only for new installation of R)
#-------------------------------------------------------------------------------
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.15")

```

```{r include=TRUE}
#Gapit Source
source("http://zzlab.net/GAPIT/GAPIT.library.R")
source("http://zzlab.net/GAPIT/gapit_functions.txt")
```

```{r include=FALSE}
library(bigmemory)
require(reshape2)
library(tidyr)
library(stringr)
library(dplyr)
library(rglwidget)
library(rgl)
require(rrBLUP)
require(factoextra)
#install.packages("errorist")
#library(errorist)
```

```{r format_files, eval=FALSE, echo=FALSE}
##Genotypic Data
###Numeric Format
###Numeric Genotype file
myGD <- read.delim("data/Intermediate_Files/C10_OLA_2022_numeric.txt", head=TRUE)

###Position of Each SNP Along the Genome file
myGM <- read.table("data/Intermediate_Files/C10_OLA_2022_SNPposition.txt", head=TRUE)

##Phenotypic Data
myY <- read.table("data/Intermediate_Files/C10_OLA_2022_pheno.txt", head=TRUE)

```

GAPIT

```{r gwas_FarmCPU_numeric, echo=FALSE}

#get list of just trait IDs
traits <- colnames(myY)[2:ncol(myY)] #get trait IDs from pheno file
traits <- unique(traits) #make sure trait names are unique 

setwd('output/BLINK/C10_OLA_2022')

#Run GAPIT
  myGAPIT <- GAPIT(
    Y=myY,
    GD=myGD,
    GM=myGM,
    PCA.total=2,
    model=c("BLINK")
  )

```

```{r, clean_up}
rm(myGAPIT, myGD, myGM, myY, traits)
setwd('/Volumes/Backup_Plus/AM/New_GWAS/2023.01.27/IWG_GWAS')
```



