---
title: "Ref+MAF"
author: "Leah Treffer"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list=ls(all=TRUE))
#set current working directory
#setwd()
setwd('/Volumes/Backup_Plus/AM/New_GWAS/2023.01.27/IWG_GWAS')
```

```{r include=FALSE}
require(reshape2)
library(tidyr)
library(stringr)
library(dplyr)
library(data.table)
install.packages("gplots")
install.packages("scatterplot3d") #The downloaded link at: http://cran.r-project.org/package=scatterplot3d
library(bigmemory)
library(googlesheets4)
library(googledrive)
#library(errorist) #automatic error search
```

Look at only traits with significant SNPs and add ref+alleles to these trait tables

Reference allele was computed using Bedtools on beocat: 
% module load BEDTools #load module BEDTools/2.29.1-GCC-8.3.0 
% bedtools getfasta -fi /homes/jcrain/IWG/Thinopyrum_intermedium_V3_release/Thinopyrum_intermedium/sequences/Thinopyrum_intermedium.fasta -bed BedTools_Reference.bed -tab -name -fo FHB_Reference_Allele.bed 

```{r}
#Add Ref and Alleles to table
refall <- read.table("data/Original_Files/FHB_Reference_Allele.bed",header = FALSE, sep="\t",stringsAsFactors=FALSE, quote="") #load reference list 
colnames(refall)[1] <- "rs" #rename snp column to rs to match hapmap format

h <- read.table('data/Original_Files/FHB_Imputed.hmp.txt', header = TRUE, check.names = FALSE)
hm <- h[,c(1:4)] #get snp block  
colnames(hm)[1] <- "SNP"

rs <- as.data.frame(gsub("\\:.*","",refall$rs)) #remove characters after the snp: section 
colnames(rs)[1] <- "SNP" #rename snp column to rs to match hapmap format
rs <- cbind(refall, rs) #combine the snp block to the map block
rs <- rs[,c(3,2)] #reorder
colnames(rs)[2] <- "Ref" #rename 
hmm <- merge(rs, hm, by="SNP", all=TRUE) #merge files

rm(hm)
```

Combine output tables

```{r separatePheno}
#Use list of Significant SNPs
#read in GAPIT.Association.PVE.FarmCPU.FHBD3G.DON.csv
#read in GAPIT.Association.GWAS_Results.FarmCPU.FHBD3G.DON.csv
#change Cycle_Site_Year for each run 

# C07_SAL_2018 C07_SAL_2019 C07_SAL_2020 C10_SAL_2021 C10_SAL_2022 C10_OLA_2021 C10_OLA_2022

#get traits that had significant p.values
trait <- read.csv('data/Intermediate_Files/PVE_C10_OLA_2022.csv', head=TRUE)
traits <- trait$Trait #make table of traits
traits <- unique(traits) #make sure trait names are unique 

#set up directories
main_dir <- 'output/FarmCPU/C10_OLA_2022/GAPIT.Association.PVE.FarmCPU.' #designate directory path 
main_dirr <- 'output/FarmCPU/C10_OLA_2022/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 
```

```{r run}
#make tables

for(i in traits){
  one = (paste0(main_dir, i, '.csv'))
  one <- read.csv(one)
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  three <- semi_join(two, one, by="SNP")
  
  four <- merge(one[,c(1,3,4,6,7)], three[,c(1,2,5,7)], by="SNP")

  my_splits <- split(trait, trait$Trait)    # Split data frame in list
  temp = my_splits[[i]]  # extract data frame in [i] list element
  
  five <- merge(four[,c(1,6,2,4,3,8,7,5)], temp[,c(2:ncol(temp))], by="SNP")
  
  six <- merge(five[,c(9,1:8,12,15)], hmm[,c(1:3)], by="SNP")

  ii = paste0(i,'_allele')
  assign(paste0(ii), six[,c(1:4,12,13,5:11)]) #assign the table the trait name of i 
}

#output is a separate data.frame table for each trait with SNP, MAF, effect, and PVE 
rm(main_dir,main_dirr,one,two,three,four,five,six,my_splits,temp,i,ii)
```

```{r}
# Reassemble into one table
beep <- paste0(traits,'_allele')
print(beep)

boop <- rbind(FHBZEA_allele,FHBDON_allele,FHBSPKINC_allele,PTHT_allele,ZDK_allele,BLSSEV_allele,HSATIVLSEV_allele) #add in all traits printed
boop$Plot <- "C10_OLA_2022"

write.csv(boop, 'data/Final_Files/QTL_C10_OLA_2022.csv')

rm(beep,boop) #remove tables
rm(FHBZEA_allele,FHBDON_allele,ZDK_allele,BLSSEV_allele,HSATIVLSEV_allele) #remove list of traits printed 
```
