---
title: "Allele_Frequencies"
author: "Leah Treffer"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list=ls(all=TRUE))
#set current working directory
#setwd()
setwd('/Volumes/Backup_Plus/AM/New_GWAS/2023.01.27/IWG_GWAS')
```

```{r include=FALSE}
require(reshape2)
library(tidyr)
library(stringr)
library(dplyr)
library(data.table)
install.packages("gplots")
install.packages("scatterplot3d") #The downloaded link at: http://cran.r-project.org/package=scatterplot3d
library(bigmemory)
library(googlesheets4)
library(googledrive)
#library(errorist) #automatic error search
```

Allelic Frequency:

# Objective

Use geno files and Ref Allele list retrieved from jcrain beocat to determine which allele is major and minor in numeric format geno file 

## Notes

this is more of a test to see if i can do it


```{r import}
#Add Ref and Alleles to table
refall <- read.table("data/Original_Files/FHB_Reference_Allele.bed",header = FALSE, sep="\t",stringsAsFactors=FALSE, quote="") #load reference list 
colnames(refall)[1] <- "rs" #rename snp column to rs to match hapmap format

hapmap <- read.table('data/Original_Files/FHB_Imputed.hmp.txt', header = TRUE, check.names = FALSE)
hm <- hapmap[,c(1:4)] #get snp block  
colnames(hm)[1] <- "SNP"

rs <- as.data.frame(gsub("\\:.*","",refall$rs)) #remove characters after the snp: section 
colnames(rs)[1] <- "SNP" #rename snp column to rs to match hapmap format
rs <- cbind(refall, rs) #combine the snp block to the map block
rs <- rs[,c(3,2)] #reorder
colnames(rs)[2] <- "Ref" #rename 
hmm <- merge(rs, hm, by="SNP", all=TRUE) #merge files

rm(hm)

geno <- read.table(file = 'data/Original_Files/FHB_for_GS_GWAS.txt', header = TRUE, row.names = 1, stringsAsFactors = FALSE) #load genotype calls

plants <- read.csv(file = 'data/Original_Files/plants.csv') #load ID list
plant <- plants[str_detect((plants$source_id), "20SGH"),] #keep only data from one population
plant <- plant[str_detect((plant$experiment_id), "20_OLA"),] #keep only data from one site

trait <- read.table('data/Intermediate_Files/SigMarkerlist_C10.txt') #input list of all SNPs that were significant (all years)
trait <- trait[str_detect((trait$Plot), "C10_OLA"),] #keep only data from one site
```

```{r format}
#Reference Alleles
#make table of snp, ref, and alleles 
rss <- hmm[,c(1:3)]

#Hapmap remove unnecessary columns, get just C7s
hapm <- hapmap[,c(1,12:ncol(hapmap))]
hh <- hapm[,c(2:ncol(hapm))]
a <- hapm[,c(1)]

#get map block for just C7 population
hh <- hh[ , grep("17SGH", colnames(hh))] 
#OR for C10's : 
p <- c(na.omit(unique(plant$source_id))) #in C10 pop, get only one site
names.use <- names(hh)[(names(hh) %in% p)] # make list of names from p that are in hh
hh <- hh[, names.use] #subset

hpmp <- cbind(a, hh) #combine the snp block to the map block
rownames(hpmp) <- hpmp$a #make first column into row names
hpmp <- hpmp[,-1] #remove first column 

#Get hapmap in same orientation as geno file
#get just the SNPs that are in our QTL set, this will reduce the size of files 
list <- unique(trait$SNP) #remove duplicate snps so each is unique 

#C07 geno
g <- as.data.frame(rownames(geno))
#OR for C10's
g <- geno[rownames(geno) %in% p, ]
g <- as.data.frame(rownames(g))

colnames(g)[1] <- "taxa"
new <- g 


#loop each snp that is in our QTL list into it's own row, transpose, and put table back together by merging 
for (j in list){ 
  row <- hpmp[j,]
  data2 <- tibble::rownames_to_column(row, "row_names") #convert row names into first column of data
  data2 <- rbind(colnames(data2), data2) #make copy of header columns and adds them as the first row
  colnames(data2) <- paste0('V', seq_len(ncol(data2))) #changes header to 'V1..'
  data2 <- as.data.frame(t(data2)) # flip table 
  colnames(data2) <- data2[1, ] #make first row into header
  data2 = data2[-1,] #remove first row
  colnames(data2)[1] <- "taxa" 
  new <- merge(data2, new, by="taxa", all=TRUE) #merge files
}

#clean up
rm(data2, a, row)
```

```{r convert}
#convert coded hapmap format to specific allele
#Genotype:Code
## AA:A
## CC:C
## GG:G
## TT:T
## AG:R
## CT:Y
## CG:S
## AT:W
## GT:K
## AC:M
## missing:N

#make copy of new to test
test <- new
#replace one letter code with the genotype 
test[test == "A"] <- "AA" #replace
test[test == "C"] <- "CC" #replace
test[test == "G"] <- "GG" #replace
test[test == "T"] <- "TT" #replace
test[test == "R"] <- "AG" #replace
test[test == "Y"] <- "CT" #replace
test[test == "S"] <- "CG" #replace
test[test == "W"] <- "AT" #replace
test[test == "K"] <- "GT" #replace
test[test == "M"] <- "AC" #replace
test[test == "N"] <- "NN" #replace

```

```{r}
#find what allele is major for each SNP

count <- test #make copy of test to work with 
rownames(count) <- count$taxa #make first column into row names
count <- count[,-1] #remove first column

#Table with count of each SNP's allele summary 
# compute unique levels in data frame
lvls <- unique(unlist(count))
# apply the summation per value 
freq <- sapply(count, 
               function(x) table(factor(x, levels = lvls, 
                                        ordered = TRUE)))

#Table of SNPs, Reference, Alleles, and Frequencies 
freq2 <- as.data.frame(t(freq)) #transpose and change to dataframe
freq2 <- tibble::rownames_to_column(freq2, "rs") #convert row names into first column of data
colnames(rss)[1] <- "rs"
freq2 <- merge(freq2, rss, by="rs") #merge to add in ref allele 
freq2 <- freq2[,c(1,12,13,2:11)] #reorder

freq2$total <- rowSums(freq2[,c(4:13)]) #Total taxa with genotypes

#Genotype Frequency = AA / Total = (# of individuals with genotype) / (total individuals)
Genotypefrequency <- cbind(freq2,
  fCT <- ((freq2$CT) / (freq2$total)),
  fCC <- ((freq2$CC) / (freq2$total)),
  fTT <- ((freq2$TT) / (freq2$total)),
  fAT <- ((freq2$AT) / (freq2$total)),
  fAA <- ((freq2$AA) / (freq2$total)),
  fAG <- ((freq2$AG) / (freq2$total)),
  fGG <- ((freq2$GG) / (freq2$total)),
  fCG <- ((freq2$CG) / (freq2$total)),
  fAC <- ((freq2$AC) / (freq2$total)),
  fGT <- ((freq2$GT) / (freq2$total))
)
#change column names
colnames(Genotypefrequency)[15] <- "fCT"
colnames(Genotypefrequency)[16] <- "fCC"
colnames(Genotypefrequency)[17] <- "fTT"
colnames(Genotypefrequency)[18] <- "fAT"
colnames(Genotypefrequency)[19] <- "fAA"
colnames(Genotypefrequency)[20] <- "fAG"
colnames(Genotypefrequency)[21] <- "fGG"
colnames(Genotypefrequency)[22] <- "fCG"
colnames(Genotypefrequency)[23] <- "fAC"
colnames(Genotypefrequency)[24] <- "fGT"

#Allelic Frequency (AF)
#GF= genotype frequencies
GF <- Genotypefrequency #create copy with shorter name
#Allelic Frequency = genomic frequency of homozygous + 1/2 genomic frequency heterozygous for specific Base  
#AF= GF(XX) + 1/2GF(Xx) 
MAfrequency <- cbind(freq2,
fA <- ((GF$fAA + (GF$fAT / 2) + (GF$fAC / 2) + (GF$fAG / 2))),
fT <- ((GF$fTT + (GF$fAT / 2) + (GF$fCT / 2) + (GF$fGT / 2))),
fG <- ((GF$fGG + (GF$fAG / 2) + (GF$fCG / 2) + (GF$fGT / 2))),
fC <- ((GF$fCC + (GF$fCT / 2) + (GF$fAC / 2) + (GF$fCG / 2)))
)
#change column names
colnames(MAfrequency)[15] <- "A"
colnames(MAfrequency)[16] <- "T"
colnames(MAfrequency)[17] <- "G"
colnames(MAfrequency)[18] <- "C"

freqs <- merge(Genotypefrequency,MAfrequency[,c(1,15:18)], by="rs")
colnames(freqs)[1] <- "SNP"
colnames(freqs)[15] <- "genotypic_frequency CT"
colnames(freqs)[16] <- "genotypic_frequency CC"
colnames(freqs)[17] <- "genotypic_frequency TT"
colnames(freqs)[18] <- "genotypic_frequency AT"
colnames(freqs)[19] <- "genotypic_frequency AA"
colnames(freqs)[20] <- "genotypic_frequency AG"
colnames(freqs)[21] <- "genotypic_frequency GG"
colnames(freqs)[22] <- "genotypic_frequency CG"
colnames(freqs)[23] <- "genotypic_frequency AC"
colnames(freqs)[24] <- "genotypic_frequency GT"
colnames(freqs)[25] <- "allelic_frequency A"
colnames(freqs)[26] <- "allelic_frequency T"
colnames(freqs)[27] <- "allelic_frequency G"
colnames(freqs)[28] <- "allelic_frequency C"

write.csv(freqs, 'data/Final_Files/Frequencies_C10_OLA.csv')

rm(count,freq,freq2,freqs,g,Genotypefrequency,GF,MAfrequency,lvls,names.use,p,j,fA,fAA,fAC,fAG,fAT,fC,fCC,fCG,fCT,fG,fGG,fGT,fT,fTT,trait,list,test,rss,hh,hpmp,new,plant)
```

#######################################NOTES#########################################################

```{r notes}
###########################################################################################################
#Genotype Frequency = AA / Total = (# of individuals with genotype) / (total individuals)
#Allele Frequency = BB + 1/2 Bb = (genotype frequency of homozygous + (1/2 genotype frequency hetereozygous))
#Frequency after Random Mating = p^2 + 2pq + q^2
  # p^2 = (allelic frequency homozygous)^2
  # 2pq = 2(allelic frequency heterozygous)
  # q^2 = (allelic frequency homozygous)^2
    #resulting allele frequency of p = p^2 + 1/2(2pq) = p^2 + pq 
    #resulting allele frequency of q= q^2 + 1/2(2pq) = q^2 + pq

#Two ways to get Total excluding the no counts (NN):
# 1. eliminate no calls
# Total = total - count of NN = (458-(freq2$NN)))
# Total - total Bases - NN Bases = (916 - (freq2$NN * 2)))
# 2. Add genotypes only 
# Total = Sum of counts of CT,AC,etc. = ((rowSums(freq2[,c(4:13)]))))
# Total = Sum of Bases occuring in CT,AC,etc. = ((rowSums(freq2[,c(4:13)]))*2))

#Get Allelic Frequency from Count instead of from Genomic Frequency: 
#Allelic Frequency when freq2 = count of individuals with certain genotype, eliminate no calls in denominator;  here each Base counts as an allele instance (ex. homozygous has two instances of the Base)
AF <- cbind(freq2,
fA <- (((freq2$AA*2) + freq2$AT + freq2$AC + freq2$AG)  / (916 - (freq2$NN * 2))),
fT <- (((freq2$TT*2) + freq2$AT + freq2$CT + freq2$GT)  / (916 - (freq2$NN * 2))),
fG <- (((freq2$GG*2) + freq2$AG + freq2$CG + freq2$GT)  / (916 - (freq2$NN * 2))),
fC <- (((freq2$CC*2) + freq2$CT + freq2$AC + freq2$CG)  / (916 - (freq2$NN * 2)))
)

#Allelic Frequency when freq2 = count of individuals with certain genotype, not using no calls as part of total count in denomintor, here it is instince of allele within each haplotype (ex. Base occurs in half of heterozygous genotype)
AF <- cbind(freq2,
fA <- ((freq2$AA + (freq2$AT / 2) + (freq2$AC / 2) + (freq2$AG / 2))  / ((rowSums(freq2[,c(4:13)])))),
fT <- ((freq2$TT + (freq2$AT / 2) + (freq2$CT / 2) + (freq2$GT / 2))  / ((rowSums(freq2[,c(4:13)])))),
fG <- ((freq2$GG + (freq2$AG / 2) + (freq2$CG / 2) + (freq2$GT / 2))  / ((rowSums(freq2[,c(4:13)])))),
fC <- ((freq2$CC + (freq2$CT / 2) + (freq2$AC / 2) + (freq2$CG / 2))  / ((rowSums(freq2[,c(4:13)]))))
)

#Allelic Frequency when freq2 = count of individuals with certain genotype, not using no calls as part of total count in denomintor; here each bp counts as an allele instance 
MAfrequency <- cbind(freq2,
fA <- (((freq2$AA*2) + freq2$AT + freq2$AC + freq2$AG)  / ((rowSums(freq2[,c(4:13)]))*2)),
fT <- (((freq2$TT*2) + freq2$AT + freq2$CT + freq2$GT)  / ((rowSums(freq2[,c(4:13)]))*2)),
fG <- (((freq2$GG*2) + freq2$AG + freq2$CG + freq2$GT)  / ((rowSums(freq2[,c(4:13)]))*2)),
fC <- (((freq2$CC*2) + freq2$CT + freq2$AC + freq2$CG)  / ((rowSums(freq2[,c(4:13)]))*2))
)

#frequency after random mating
AF <- MAfrequency
#AF= Allelic frequency 
RMfrequency <- cbind(freq2,
rmA <- (((AF$A)^2) + (1/2(2(AF$A)(AF$T)(AF$G)(AF$C)))),
rmT <- (((AF$T)^2) + (1/2(2(AF$A)(AF$T)(AF$G)(AF$C)))),
rmG <- (((AF$G)^2) + (1/2(2(AF$A)(AF$T)(AF$G)(AF$C)))),
rmC <- (((AF$C)^2) + (1/2(2(AF$A)(AF$T)(AF$G)(AF$C))))
)
```
