---
title: "FHB1 Region"
output: html_document
date: "2022-10-27"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list=ls(all=TRUE))
#set current working directory
#setwd()
setwd('/Volumes/Backup_Plus/AM/New_GWAS/2023.01.27/IWG_GWAS')
```

```{r}
library(stringr)
install.packages("bindr")
library(bindr)
library(dplyr)
#install.packages("errorist")
#library(errorist)
```

# Objective: Determine if the FHB1 region has an effect on FHB traits using a priori hypothesis. 
  # If the region has no effect then a p.value calculated to just the region instead of multiple testing (like in GWAS) then will be null. 
  # If the region does have an effect then markers will have a significant p.value
# 1. How many markers are in FHB1 region (What is the effective marker number)
# 2. Calculate new P value
# 3. Reevaluate output of FHB related traits using new p.value threshold as the cutoff


```{r}
#Markers after filtering out for MAF, missing, monomorphic
FHB_for_GS_GWAS <- read.table(file = 'data/Original_Files/FHB_for_GS_GWAS.txt', header = TRUE, row.names = 1, stringsAsFactors = FALSE) #load genotype calls
#make a copy of master file to work with
GENO <- FHB_for_GS_GWAS
geno <- GENO
#remove unstructured chromosomes from mapping 245 markers
geno <- geno[, !grepl('SUN', colnames(geno), ignore.case = TRUE) ] 
geno <- cbind(rownames(geno), data.frame(geno, row.names=NULL)) #rownames into first column 
colnames(geno)[1]  <-  "taxa" 
###Position of Each SNP Along the Genome file
#GWAS fit geno file for gwas
rr_gwas <- t(geno) #transpose
#format marker names
markers <- as.data.frame(rownames(rr_gwas)) #extract maker names
colnames(markers)[1] <- 'marker_name' #rename first column 
markers$pos <- gsub('^.*_', '', markers$marker_name) #split string and extract marker position
markers$pos <- as.numeric(markers$pos) #convert to numeric
markers$chrom <- gsub('_.*$', '', markers$marker_name) #split string and extract chromosome
markers$chrom <- sub('S', '', markers$chrom) #remove string 
markers <- markers[order(markers$chrom, markers$pos),] #orders markers by chromosome and position
markers$marker <- 1:nrow(markers) #add marker name as numeric
markers <- markers[ , c(4:1)] #rearrange columns
markers <- subset(markers, chrom != 'taxa') #find 'taxa' row and remove
GM <- markers[, c(4,2,3)] #move to order required for GAPIT hapmap (Name, chromosome, position)
colnames(GM)[1]  <-  "Name" 
colnames(GM)[2]  <-  "Chromosome" 
colnames(GM)[3]  <-  "Position" 
write.table(GM, file = 'data/Intermediate_Files/SNP_Position.txt', sep = '\t', quote = FALSE, row.names = FALSE)
GM <- read.table('data/Intermediate_Files/SNP_Position.txt', head=TRUE)

rm(GENO,geno,rr_gwas,markers)
```


```{r}
# 1. How many markers are in FHB1 region
  # J03 : J03:50676750..51198852
  # V03 : V03:18027299..18318355
  # S03 : S03:43496709..43892886

#Make separate list for each subgenome
J03 <- GM[str_detect(GM$Chromosome, "J03"), ] #extract rows with a partial match using the stringr package
V03 <- GM[str_detect(GM$Chromosome, "V03"), ] #extract rows with a partial match using the stringr package
S03 <- GM[str_detect(GM$Chromosome, "S03"), ] #extract rows with a partial match using the stringr package

#selection of rows based on range of value using subset function
J03$Position <- as.numeric(J03$Position) #needs to be numeric for filtering instead of character
J03 <- subset(J03, J03$Position>50676750 & J03$Position<51198852)

V03$Position <- as.numeric(V03$Position) #needs to be numeric for filtering instead of character
V03 <- subset(V03, V03$Position>18027299 & V03$Position<18318355)

S03$Position <- as.numeric(S03$Position) #needs to be numeric for filtering instead of character
S03 <- subset(S03, S03$Position>43496709 & S03$Position<43892886)

JS <- data.frame(Name="FHB1_start", Chromosome="J03", Position="50676750")
FHB1_new <- rbind(JS,J03)
JE <- data.frame(Name="FHB1_end", Chromosome="J03", Position="51198852")
FHB1_new <- rbind(FHB1_new,JE)
VS <- data.frame(Name="FHB1_start", Chromosome="V03", Position="18027299")
FHB1_new <- rbind(FHB1_new,VS)
FHB1_new <- rbind(FHB1_new,V03)
VE <- data.frame(Name="FHB1_end", Chromosome="V03", Position="18318355")
FHB1_new <- rbind(FHB1_new,VE)
SS <- data.frame(Name="FHB1_start", Chromosome="S03", Position="43496709")
FHB1_new <- rbind(FHB1_new,SS)
FHB1_new <- rbind(FHB1_new,S03)
SE <- data.frame(Name="FHB1_end", Chromosome="S03", Position="43892886")
FHB1_new <- rbind(FHB1_new,SE)

write.csv(FHB1_new, 'data/Intermediate_Files/FHB1_markers.csv')
rm(JS,JE,VS,VE,SS,SE,FHB1_new)
FHB1 <- read.csv('data/Intermediate_Files/FHB1_markers.csv')
```

```{r}
# 2. Calculate new P value
  # p.value threshold = (p.value) / (# markers)

# J03 : 11 markers
0.10 / 11 
0.05 / 11
0.01 / 11
# = 0.009090909
# = 0.004545455
# = 0.0009090909
# V03 : 5 markers
0.10 / 5
0.05 / 5
0.01 / 5
# = 0.02
# = 0.01
# = 0.002
# S03 : 3 markers
0.10 / 3
0.05 / 3
0.01 / 3
# = 0.03333333
# = 0.01666667
# = 0.003333333
```

```{r}
# 3. Reevaluate output of FHB related traits using new p.value threshold as the cutoff
# Do by year 2018, 2019, 2020, 2021, 2022
# 2018 : FHBD3G.DON, FHBDISIND, FHBDON, FHBSPKINC
# 2019 : FHBD3G.DON, FHBD3G, FHBDISIND, FHBDON, FHBPKINC
# 2020 : FHBDISIND, FHBDON, FHBPKINC
# 2021 : FHBDON, FHBDISIND, FHBSPKINC
# 2022 : FHBZEA, FHBDON, FHBDISIND, FHBSPKINC

#2018
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBD3G.DON", "FHBDISIND", "FHBDON", "FHBSPKINC", "FHBD3G")
main_dirr <- 'output/FarmCPU/C07_SAL_2018/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)
  
  i = paste0('2018_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9, 1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#2019
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBD3G.DON", "FHBDISIND", "FHBDON", "FHBSPKINC", "FHBD3G")
main_dirr <- 'output/FarmCPU/C07_SAL_2019/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)
  
  i = paste0('2019_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9, 1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#2020
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBDISIND", "FHBDON", "FHBSPKINC")
main_dirr <- 'output/FarmCPU/C07_SAL_2020/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)

  i = paste0('2020_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9,1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#OLA 2021
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBDISIND", "FHBDON", "FHBSPKINC")
main_dirr <- 'output/FarmCPU/C10_OLA_2021/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)

  i = paste0('2021O_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9,1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#OLA 2022
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBZEA", "FHBDON", "FHBDISIND", "FHBSPKINC")
main_dirr <- 'output/FarmCPU/C10_OLA_2022/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)

  i = paste0('2022O_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9,1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#SAL 2021
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBDISIND", "FHBDON", "FHBSPKINC")
main_dirr <- 'output/FarmCPU/C10_SAL_2021/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)

  i = paste0('2021S_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9,1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#SAL 2022
#Set up and run loop to read in output tables for disease related traits and assign a year_trait name
traits <- c("FHBZEA", "FHBDON", "FHBDISIND", "FHBSPKINC")
main_dirr <- 'output/FarmCPU/C10_SAL_2022/GAPIT.Association.GWAS_Results.FarmCPU.' #designate directory path 

for(i in traits){
  two = (paste0(main_dirr, i, '.csv'))
  two <- read.csv(two)
  j <- two[str_detect(two$Chr, "J03"), ] #extract rows with a partial match using the stringr package
  j <- subset(j, j$Pos>50676750 & j$Pos<51198852)
  v <- two[str_detect(two$Chr, "V03"), ] #extract rows with a partial match using the stringr package
  v <- subset(v, v$Pos>18027299 & v$Pos<18318355)
  s <- two[str_detect(two$Chr, "S03"), ] #extract rows with a partial match using the stringr package
  s <- subset(s, s$Pos>43496709 & s$Pos<43892886)
  two <- bind_rows(j,v,s)

  i = paste0('2022S_', i)
  two$trait = paste(i)
  
  assign(paste0(i), two[,c(9,1:8)]) #assign the reordered four table the trait name of i 
}
rm(two, i, traits, main_dirr, j, v, s)

#Clean up lists to just the Chr 3 range we are interested in 
all2018 <- bind_rows(`2018_FHBD3G.DON`, `2018_FHBDISIND`, `2018_FHBDON`, `2018_FHBSPKINC`, `2018_FHBD3G`)
all2019 <- bind_rows(`2019_FHBD3G.DON`, `2019_FHBD3G`, `2019_FHBDISIND`, `2019_FHBDON`, `2019_FHBSPKINC`)
all2020 <- bind_rows(`2020_FHBDISIND`, `2020_FHBDON`, `2020_FHBSPKINC`)
allO2021 <- bind_rows(`2021O_FHBDISIND`,`2021O_FHBDON`,`2021O_FHBSPKINC`)
allO2022 <- bind_rows(`2022O_FHBZEA`,`2022O_FHBDON`,`2022O_FHBDISIND`,`2022O_FHBSPKINC`)
allS2021 <- bind_rows(`2021S_FHBDISIND`,`2021S_FHBDON`,`2021S_FHBSPKINC`)
allS2022 <- bind_rows(`2022S_FHBZEA`,`2022S_FHBDON`,`2022S_FHBDISIND`,`2022S_FHBSPKINC`)


rm(`2018_FHBD3G`, `2018_FHBD3G.DON`, `2018_FHBDISIND`, `2018_FHBDON`, `2018_FHBSPKINC`,`2019_FHBD3G.DON`, `2019_FHBD3G`, `2019_FHBDISIND`, `2019_FHBDON`, `2019_FHBSPKINC`,`2020_FHBDISIND`, `2020_FHBDON`, `2020_FHBSPKINC`, `2021O_FHBDISIND`,`2021O_FHBDON`,`2021O_FHBSPKINC`, `2022O_FHBDISIND`,`2022O_FHBSPKINC`, `2021S_FHBDISIND`,`2021S_FHBDON`,`2021S_FHBSPKINC`, `2022S_FHBDISIND`,`2022S_FHBSPKINC`)
```

```{r}
#Subset rows based on p value using subset function
#J03
# P Value threshold = 0.009090909
# P Value threshold = 0.004545455
# P Value threshold = 0.0009090909
#V03
# P Value threshold = 0.02
# P Value threshold = 0.01
# P Value threshold = 0.002
#S03
# P Value threshold = 0.03333333
# P Value threshold = 0.01666667
# P Value threshold = 0.003333333

# P.Value Threshold of 0.10
threshold_0.10_2018 <- subset(all2018, all2018$Chr == "J03" & all2018$P.value<0.009090909 | all2018$Chr == "V03" & all2018$P.value<0.02 | all2018$Chr == "S03" & all2018$P.value<0.03333333) #extract rows with p.values less than J03 threshold, or V03 threshold, or S03 threshold 
threshold_0.10_2019 <- subset(all2019, all2019$Chr == "J03" & all2019$P.value<0.009090909 | all2019$Chr == "V03" & all2019$P.value<0.02 | all2019$Chr == "S03" & all2019$P.value<0.03333333)
threshold_0.10_2020 <- subset(all2020, all2020$Chr == "J03" & all2020$P.value<0.009090909 | all2020$Chr == "V03" & all2020$P.value<0.02 | all2020$Chr == "S03" & all2020$P.value<0.03333333)
threshold_0.10_O2021 <- subset(allO2021, allO2021$Chr == "J03" & allO2021$P.value<0.009090909 | allO2021$Chr == "V03" & allO2021$P.value<0.02 | allO2021$Chr == "S03" & allO2021$P.value<0.03333333)
threshold_0.10_O2022 <- subset(allO2022, allO2022$Chr == "J03" & allO2022$P.value<0.009090909 | allO2022$Chr == "V03" & allO2022$P.value<0.02 | allO2022$Chr == "S03" & allO2022$P.value<0.03333333)
threshold_0.10_S2021 <- subset(allS2021, allS2021$Chr == "J03" & allS2021$P.value<0.009090909 | allS2021$Chr == "V03" & allS2021$P.value<0.02 | allS2021$Chr == "S03" & allS2021$P.value<0.03333333)
threshold_0.10_S2022 <- subset(allS2022, allS2022$Chr == "J03" & allS2022$P.value<0.009090909 | allS2022$Chr == "V03" & allS2022$P.value<0.02 | allS2022$Chr == "S03" & allS2022$P.value<0.03333333)

# P.Value Threshold of 0.05
threshold_0.05_2018 <- subset(all2018, all2018$Chr == "J03" & all2018$P.value<0.004545455 | all2018$Chr == "V03" & all2018$P.value<0.01 | all2018$Chr == "S03" & all2018$P.value<0.01666667) #extract rows with p.values less than J03 threshold, or V03 threshold, or S03 threshold 
threshold_0.05_2019 <- subset(all2019, all2019$Chr == "J03" & all2019$P.value<0.004545455 | all2019$Chr == "V03" & all2019$P.value<0.01 | all2019$Chr == "S03" & all2019$P.value<0.01666667)
threshold_0.05_2020 <- subset(all2020, all2020$Chr == "J03" & all2020$P.value<0.004545455 | all2020$Chr == "V03" & all2020$P.value<0.01 | all2020$Chr == "S03" & all2020$P.value<0.01666667)
threshold_0.05_O2021 <- subset(allO2021, allO2021$Chr == "J03" & allO2021$P.value<0.004545455 | allO2021$Chr == "V03" & allO2021$P.value<0.01 | allO2021$Chr == "S03" & allO2021$P.value<0.01666667)
threshold_0.05_O2022 <- subset(allO2022, allO2022$Chr == "J03" & allO2022$P.value<0.004545455 | allO2022$Chr == "V03" & allO2022$P.value<0.01 | allO2022$Chr == "S03" & allO2022$P.value<0.01666667)
threshold_0.05_S2021 <- subset(allS2021, allS2021$Chr == "J03" & allS2021$P.value<0.004545455 | allS2021$Chr == "V03" & allS2021$P.value<0.01 | allS2021$Chr == "S03" & allS2021$P.value<0.01666667)
threshold_0.05_S2022 <- subset(allS2022, allS2022$Chr == "J03" & allS2022$P.value<0.004545455 | allS2022$Chr == "V03" & allS2022$P.value<0.01 | allS2022$Chr == "S03" & allS2022$P.value<0.01666667)


# P.Value Threshold of 0.01
threshold_0.01_2018 <- subset(all2018, all2018$Chr == "J03" & all2018$P.value<0.0009090909 | all2018$Chr == "V03" & all2018$P.value<0.002 | all2018$Chr == "S03" & all2018$P.value<0.003333333) 
threshold_0.01_2019 <- subset(all2019, all2019$Chr == "J03" & all2019$P.value<0.0009090909 | all2019$Chr == "V03" & all2019$P.value<0.002 | all2019$Chr == "S03" & all2019$P.value<0.003333333)
threshold_0.01_2020 <- subset(all2020, all2020$Chr == "J03" & all2020$P.value<0.0009090909 | all2020$Chr == "V03" & all2020$P.value<0.002 | all2020$Chr == "S03" & all2020$P.value<0.003333333)
threshold_0.01_O2021 <- subset(allO2021, allO2021$Chr == "J03" & allO2021$P.value<0.0009090909 | allO2021$Chr == "V03" & allO2021$P.value<0.002 | allO2021$Chr == "S03" & allO2021$P.value<0.003333333)
threshold_0.01_O2022 <- subset(allO2022, allO2022$Chr == "J03" & allO2022$P.value<0.0009090909 | allO2022$Chr == "V03" & allO2022$P.value<0.002 | allO2022$Chr == "S03" & allO2022$P.value<0.003333333)
threshold_0.01_S2021 <- subset(allS2021, allS2021$Chr == "J03" & allS2021$P.value<0.0009090909 | allS2021$Chr == "V03" & allS2021$P.value<0.002 | allS2021$Chr == "S03" & allS2021$P.value<0.003333333)
threshold_0.01_S2022 <- subset(allS2022, allS2022$Chr == "J03" & allS2022$P.value<0.0009090909 | allS2022$Chr == "V03" & allS2022$P.value<0.002 | allS2022$Chr == "S03" & allS2022$P.value<0.003333333)

#threshold_0.10_2018$threshold <- "0.10_2018"
#threshold_0.10_2019$threshold <- "0.10_2019"
#threshold_0.10_2020$threshold <- "0.10_2020"
#threshold_0.10_O2021$threshold <- "0.10_O_2021"
threshold_0.10_O2022$threshold <- "0.10_O_2022"
#threshold_0.10_S2021$threshold <- "0.10_S_2021"
#threshold_0.10_S2022$threshold <- "0.10_S_2022"

#threshold_0.05_2018$threshold <- "0.05_2018"
#threshold_0.05_2019$threshold <- "0.05_2019"
#threshold_0.05_2020$threshold <- "0.05_2020"
#threshold_0.05_O2021$threshold <- "0.05_O_2021"
threshold_0.05_O2022$threshold <- "0.05_O_2022"
#threshold_0.05_S2021$threshold <- "0.05_S_2021"
#threshold_0.05_S2022$threshold <- "0.05_S_2022"

#threshold_0.01_2018$threshold <- "0.01_2018"
#threshold_0.01_2019$threshold <- "0.01_2019"
#threshold_0.01_2020$threshold <- "0.01_2020"
#threshold_0.01_O2021$threshold <- "0.01_O_2021"
threshold_0.01_O2022$threshold <- "0.01_O_2022"
#threshold_0.01_S2021$threshold <- "0.01_S_2021"
#threshold_0.01_S2022$threshold <- "0.01_S_2022"

threshold <- rbind(threshold_0.10_O2022,threshold_0.05_O2022,threshold_0.01_O2022)

```

```{r}
# Region Example V01:504888459..505003458 
threshold$LPos <- threshold$Pos - 100
threshold$HPos <- threshold$Pos + 100

threshold$Phytozome <- paste(threshold$Chr,":",threshold$LPos,"..",threshold$HPos)

threshold$T.inter_Genes <- c("NA", "Thintv31194873m.g", "NA", "Thintv31194873m.g", "Thintv31194873m.g", "Thintv31047566m.g", "NA", "Thintv31194873m.g", "NA", "Thintv31047566m.g", "Thintv31047566m.g")
threshold$T.inter_Description <- c("NA", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN", "NA", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN", "NA", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN", "NA", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN", "Alpha-L-fucosidase ; GDSL-like Lipase/Acylhydrolase ; ZINC FINGER FYVE DOMAIN CONTAINING PROTEIN")

threshold$NCBI_Description <- c("uncharacterized ; putative pentatricopeptide repeat-containing protein", "Sumai Fhb1 genomic region", "NA", "Sumai Fhb1 genomic region", "Sumai Fhb1 genomic region", "NA", "uncharacterized ; putative pentatricopeptide repeat-containing protein", "Sumai Fhb1 genomic region", "NA", "NA", "NA")
threshold$NCBI_Species <- c("T.aestivum", "T.aestivum", "NA", "T.aestivum", "T.aestivum", "NA", "T.aestivum", "T.aestivum", "NA", "NA", "NA")
threshold$Query <- c("88% ; 80%", "89%", "NA", "74%", "74%", "NA", "88% ; 80%", "89%", "NA", "NA", "NA")
threshold$Per.Ident <- c("91.01% ; 86.55%", "95%", "NA", "95.97%", "95.97%", "NA", "91.01% ; 86.55%", "95%", "NA", "NA", "NA")

write.csv(threshold, 'data/Final_Files/priori_sig.markers_adjusted_threshold.csv')

```

```{r}
# Another way to try looking for markers above threshold: 
#combine lists of markers in region and then semi_join with all2018 to see if any are in both lists
mega <- bind_rows(J03, V03, S03) #combind lists of each chro into long list
colnames(mega)[1] <- "SNP" #rename marker as SNP
try <- semi_join(all2018, mega, by="SNP") #return rows in all2018 that are in the mega list
#sort by chr and visually look at P. values 
try <- semi_join(all2019, mega, by="SNP") #return rows in all2018 that are in the mega list
try <- semi_join(all2020, mega, by="SNP") #return rows in all2018 that are in the mega list

rm(try)
rm(mega)
```

```{r clean_up}
rm(threshold_0.05_O2022, threshold_0.05_O2021, threshold_0.05_S2022, threshold_0.05_S2021, threshold_0.05_2020, threshold_0.05_2019, threshold_0.05_2018, threshold_0.01_2018, threshold_0.01_2019, threshold_0.01_2020, threshold_0.01_O2021, threshold_0.01_O2022, threshold_0.01_S2021, threshold_0.01_S2022, threshold_0.10_2018, threshold_0.10_2019, threshold_0.10_2020, threshold_0.10_O2021, threshold_0.10_O2022, threshold_0.10_S2021, threshold_0.10_S2022)
rm(all2018,all2019,all2020,allO2021,allO2022,allS2021,allS2022)
rm(J03,V03,S03,GM)
```