---
title: "Revision_Additions"
author: "Leah Treffer"
date: "2025-10-20"
output: html_document
---

```{r libraries, include=FALSE}
library(here)
library(rrBLUP)
#install.packages("pheatmap")
library(pheatmap)
library(factoextra)
library(stringr)
library(dplyr)
library(lmerTest)
library(emmeans)

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("SNPRelate")

library(SNPRelate)
```

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here())
```

```{r}
FHB_for_GS_GWAS <- read.table(file = 'data/Original_Files/FHB_for_GS_GWAS.txt', header = TRUE, row.names = 1, stringsAsFactors = FALSE) #load genotype calls
#make a copy of master file to work with
GENO <- FHB_for_GS_GWAS
geno <- GENO + 1 # change geno numeric format from -101 to 012

#remove unstructured chromosomes from mapping 245 markers
geno <- geno[, !grepl('SUN', colnames(geno), ignore.case = TRUE) ]
```

```{r}
#load in phenotype tables
pheno <- read.delim('data/Intermediate_Files/Phenotypic_Format_long_2.txt') #read in file with all cycles and years and locations
pheno$plant_id <- as.factor(pheno$plant_id)
pheno$phenotype_year <- as.factor(pheno$phenotype_year)
pheno$Site <- as.factor(pheno$Site)
pheno$Time <- as.factor(pheno$Time)
#create separate tables for each cycle, year, location
#Cycle 
C07 <- pheno[str_detect(pheno$Cycle, "C7"),] #keep only data from C7 population
C10 <- pheno[str_detect(pheno$Cycle, "C10"),] #keep only data from C10 population
```

# Broad Sense Heritability

How much of observed variance is due to genetics

Reviewer 2 comment:The manuscript has multiple-year evaluations (“three years from 2018-2020 for Cycle 7 and two years from 2021-2022 for Cycle 10”). Since you have replicated observations across years for each population, can you please report the broad-sense heritability for the traits? This would help assess how much of the observed variation is genetic versus environmental.

General Formula:
Broad-Sense Heritability (H^2) = genotypic variance^2 / phenotypic variance^2
$$H^2 = \frac{σ_g^2}{σ_p^2}$$

IWG trait Heritability estimate from Bajgain et al 2025 (https://onlinelibrary.wiley.com/doi/abs/10.1002/csc2.70126):
$$H^2 = \frac{σ_g^2 }{(σ_g^2 + \frac{σ_{gl}^2}{l} + \frac{σ_{gy}^2}{y} + \frac{σ_e^2}{ly})}$$

where $σ_g^2$ is the genetic variance, $σ_{gl}^2$ is the genotype × location variance, $σ_{gy}^2$ is the genotype × year variance, $σ_e^2$ is the residual variance, l is the number of locations, and y is the number of years. Variance components used to estimate $H^2$ were obtained by fitting a linear model using the aov function in R.

most genotypes aren’t repeated across sites so there was an issue where no plant_id:Site interaction was dropped from a few of the models. In those cases, variance for that interaction set to 0

####### H2 (C07 + C10) has no residual error variance !!! #########

```{r H2}
# FHBDISIND
# C07 fit linear model using aov 
C07_FHBDISIND <- pheno[str_detect(C07$trait_id, "FHBDISIND"),]
model_C07_FHBDISIND <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C07_FHBDISIND)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C07_FHBDISIND)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C07_FHBDISIND$Site)
y <- nlevels(C07_FHBDISIND$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C07_FHBDISIND <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# C10 fit linear model using aov 
C10_FHBDISIND <- pheno[str_detect(C10$trait_id, "FHBDISIND"),]
model_C10_FHBDISIND <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C10_FHBDISIND)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C10_FHBDISIND)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C10_FHBDISIND$Site)
y <- nlevels(C10_FHBDISIND$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C10_FHBDISIND <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# NEEDS WORK
H2_FHBDISIND <- pheno[str_detect(pheno$trait_id, "FHBDISIND"),]
H2_model_FHBDISIND <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=H2_FHBDISIND)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(H2_model_FHBDISIND)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(H2_FHBDISIND$Site)
y <- nlevels(H2_FHBDISIND$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_FHBDISIND <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# FHBSPKINC
# C07 fit linear model using aov 
C07_FHBSPKINC <- pheno[str_detect(C07$trait_id, "FHBSPKINC"),]
model_C07_FHBSPKINC <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C07_FHBSPKINC)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C07_FHBSPKINC)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C07_FHBSPKINC$Site)
y <- nlevels(C07_FHBSPKINC$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C07_FHBSPKINC <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# C10 fit linear model using aov 
C10_FHBSPKINC <- pheno[str_detect(C10$trait_id, "FHBSPKINC"),]
model_C10_FHBSPKINC <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C10_FHBSPKINC)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C10_FHBSPKINC)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C10_FHBSPKINC$Site)
y <- nlevels(C10_FHBSPKINC$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C10_FHBSPKINC <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# FHBDON
# C07 fit linear model using aov 
C07_FHBDON <- pheno[str_detect(C07$trait_id, "FHBDON"),]
model_C07_FHBDON <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C07_FHBDON)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C07_FHBDON)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C07_FHBDON$Site)
y <- nlevels(C07_FHBDON$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C07_FHBDON <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# C10 fit linear model using aov 
C10_FHBDON <- pheno[str_detect(C10$trait_id, "FHBDON"),]
model_C10_FHBDON <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C10_FHBDON)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C10_FHBDON)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C10_FHBDON$Site)
y <- nlevels(C10_FHBDON$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C10_FHBDON <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# FHBD3G:DON
# C07 fit linear model using aov 
C07_FHBD3G.DON <- pheno[str_detect(C07$trait_id, "FHBD3G.DON"),]
model_C07_FHBD3G.DON <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C07_FHBD3G.DON)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C07_FHBD3G.DON)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C07_FHBD3G.DON$Site)
y <- nlevels(C07_FHBD3G.DON$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C07_FHBD3G.DON <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# C10 fit linear model using aov 
C10_FHBD3G.DON <- pheno[str_detect(C10$trait_id, "FHBD3G.DON"),]
model_C10_FHBD3G.DON <- aov(phenotype_value ~ plant_id*Site + plant_id*Time, data=C10_FHBD3G.DON)
# extract the mean squares (MS) from the ANOVA table
anova_table <- summary(model_C10_FHBD3G.DON)[[1]] # extract summary stats
# Estimate variance components
MS <- setNames(anova_table[,"Mean Sq"], trimws(rownames(anova_table))) # named string with extra blank space removed from names
MSg <- unname(MS["plant_id"])
MSgl <- unname(MS["plant_id:Site"])
# MSgl <- 0 # if plant_id:Site interaction is omited from model
MSgy <- unname(MS["plant_id:Time"])
MSe <- unname(MS["Residuals"])
# number of locations and years
l <- nlevels(C10_FHBD3G.DON$Site)
y <- nlevels(C10_FHBD3G.DON$Time)
# variance components
sigma_g  <- (MSg - MSgl - MSgy + MSe) / (l*y)
sigma_gl <- (MSgl - MSe) / y
sigma_gy <- (MSgy - MSe) / l
sigma_e  <- MSe
H2_C10_FHBD3G.DON <- sigma_g^2 / (sigma_g^2 + (sigma_gl^2 / l) + (sigma_gy^2 / y) + (sigma_e / l*y))
rm(anova_table,MS,MSg,MSgl,MSgy,MSe,l,y,sigma_g,sigma_gl,sigma_gy,sigma_e)

# FHBZEA
# Only observed in one year (2021)

# H^2:
## FHBDISIND
# C07: 0.166386462161804
# C10: 0.297051012222231
## FHBSPKINC
# C07: 0.160506085066806
# C10: 0.000001168072062
## FHBDON
# C07: 0.134445416512519
# C10: 0.105843551060923
## FHBD3G.DON
# C07: 0.305932572300649
# C10: 0.442746333116891
```

# Analysis of Varience

Reviewer 2 comment: Figures 1a and 1b show differences in disease levels between C07 and C10. Are these differences statistically significant? Should it be a concern for the breeding program? Some statistical support (e.g., mixed models) would strengthen the interpretation. 

```{r}
# get both cycles for a given trait
FHBD3G.DON <- rbind(C07_FHBD3G.DON, C10_FHBD3G.DON)
FHBDISIND <- rbind(C07_FHBDISIND, C10_FHBDISIND)
FHBDON <- rbind(C07_FHBDON, C10_FHBDON)
FHBSPKINC <- rbind(C07_FHBSPKINC, C10_FHBSPKINC)

FHBD3G.DON$Cycle <- as.factor(FHBD3G.DON$Cycle)
FHBDISIND$Cycle <- as.factor(FHBDISIND$Cycle)
FHBDON$Cycle <- as.factor(FHBDON$Cycle)
FHBSPKINC$Cycle <- as.factor(FHBSPKINC$Cycle)

# Liklihood Ratio Test - Finding better fitting model
model1 <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBD3G.DON)
model2 <- lmer(phenotype_value ~ Cycle + (1|Site), data=FHBD3G.DON)
anova(model1,model2)
model1 <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBDISIND)
model2 <- lmer(phenotype_value ~ Cycle + (1|Site), data=FHBDISIND)
anova(model1,model2)
model1 <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBDON)
model2 <- lmer(phenotype_value ~ Cycle + (1|Site), data=FHBDON)
anova(model1,model2)
model1 <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBSPKINC)
model2 <- lmer(phenotype_value ~ Cycle + (1|Site), data=FHBSPKINC)
anova(model1,model2)

# mixed model - are cycles sig different from each other?
model_D3G.DON <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBD3G.DON)
summary(model_D3G.DON)
emmeans(model_D3G.DON, pairwise ~ Cycle) #estimates of the difference between cycles
# C07 and C10 are sig different (p.value = <2e-16)
# C07 has a significantly lower D3G.DON value compared with C10 by 55.678 units (after controlling for Site)
# large confidence interval indicating uncertainty, however difference was still significant (p.value <.0001)

model_FHBDISIND <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBDISIND)
summary(model_FHBDISIND)
emmeans(model_FHBDISIND, pairwise ~ Cycle) #estimates of the difference between cycles
# C07 and C10 are sig different (p.value = <2e-16)
# C07 has a significantly lower FHBDISIND value compared with C10 by 67.935 units (after controlling for Site)

model_FHBDON <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBDON)
summary(model_FHBDON)
emmeans(model_FHBDON, pairwise ~ Cycle) #estimates of the difference between cycles
# C07 and C10 are sig different (p.value = <2e-16)
# C07 has a significantly lower FHBDISIND value compared with C10 by 33.2420 units (after controlling for Site)

model_FHBSPKINC <- lmer(phenotype_value ~ Cycle * Time + (1|Site), data=FHBSPKINC)
summary(model_FHBSPKINC)
emmeans(model_FHBSPKINC, pairwise ~ Cycle) #estimates of the difference between cycles
# C07 and C10 are sig different (p.value = <2e-16)
# C07 has a significantly higher FHBSPKINC value compared with C10 by 24.476 units (after controlling for Site)
```

# Population Structure 

Reviewer Comment: The authors' explanation is insufficient. This requires much deeper investigation. For instance, were sensitivity analyses performed with different GWAS models, kinship matrices, or subsets of data? Could these represent strong linkage disequilibrium (LD) blocks containing major-effect genes, or are they statistical artifacts? 

While FarmCPU was used, there is minimal discussion on population structure, LD decay, or kinship matrix, which could influence false-positive associations. Please provide additional analysis or information about the population structure and LD decay to better contextualize the reliability of the GWAS hits.

The FarmCPU model, as stated, is designed to control for both fixed (Q matrix, PCs) and random (K matrix, kinship) effects. The authors must clarify if a kinship matrix was incorporated into their FarmCPU model, and if not, justify this omission or acknowledge it as a major limitation.

* LD is analyzed in 0.7_Linkage_Disequalibrium.Rmd 

# PCA for Population Structure

```{r pca}
# Convert your genotype to GDS format
snpgdsVCF2GDS("data/Original_Files/FHB_Imputed_sorted.vcf", "FHB_Imputed_sorted.gds", method="biallelic.only")
genofile <- snpgdsOpen("FHB_Imputed_sorted.gds")

# principal component analysis (PCA)
pca <- snpgdsPCA(genofile, autosome.only=FALSE) #PCA
pc.percent <- pca$varprop*100 #proportions to percentages
# autosome.only=FALSE : all chromosomes are used
# pca$eigenvect contains the coordinates of each sample on the PCs
# pca$varprop gives the proportion of total genetic variance explained by each PC

plot(pca$eigenvect[,1], pca$eigenvect[,2], 
     xlab=paste0("PC1 (", round(pc.percent[1],1), "%)"),
     ylab=paste0("PC2 (", round(pc.percent[2],1), "%)"),
     pch=19, col="blue")

# PC1 (first principal component) : explains the largest variance in your genotypes
# PC2 : explains the second largest variance
# Each point is one genotype (individual/sample)
# Clusters of points = samples with similar genetic backgrounds
# Distance between points : genetic differentiation (more distant = more genetically different)
# % in axis labels : how much of total genetic variance captured by that PC
# changing pca$eigenvect[,1] and pca$eigenvect[,2] changes the PC plotted

# PC1 capturing 2.3% of total varience is low. This could be due to a highly diverse dataset and/or no strong population structure
# gradual differentiation rather than discrete subpopulations
```

```{r pca2, echo=FALSE}
# Another way to visualize Pop Structure / PC

#get realized relationship matrix
A <- A.mat(geno) #estimate the realized relationship matrix (Gmatrix)

#PCA on genotype matrix
prmarker <- prcomp(geno)

fviz_eig(prmarker, addlabels = TRUE, title= 'Eigen Value') #Plot the eigenvalues/variances against the number of dimensions
#Eigenvalues correspond to the amount of the variation explained by each principal component (PC).

#look for population structure
eig.result <- eigen(A) #get Eigen values from the A matrix
lambda <- eig.result$values #extract variance explained

rm(A,eig.result,prmarker,lambda)
```

# Kinship Matrix 

```{r}
# Using Base R
K <- A.mat(as.matrix(geno))  # additive relationship matrix

any(is.na(K))       
any(is.nan(K))
any(is.infinite(K))
K[is.na(K)] <- 0
K[is.nan(K)] <- 0

#heatmap(K, symm = TRUE, main = "Kinship matrix") 
# Error: C stack usage  7953720 is too close to the limit

image(K, axes=FALSE, col = heat.colors(100))
title("Kinship matrix")
```

```{r}
# Using GAPIT to visualize

# GAPIT - Genomic Association and Prediction Integrated Tool
# Designed by Zhiwu Zhang
# Written by Zhiwu Zhang, Alex Lipka, Feng Tian, You Tang and Jiabo Wang
# Last update: Nov 15, 2021 for version 3

#Install packages (Do this section only for new installation of R)
#-------------------------------------------------------------------------------
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.18")

#Gapit Source
source("http://zzlab.net/GAPIT/GAPIT.library.R")
source("http://zzlab.net/GAPIT/gapit_functions.txt")
```

```{r}
# Using GAPIT to visualize
genotype_data <- cbind(rownames(geno), data.frame(geno, row.names=NULL)) #rownames into first column 
colnames(genotype_data)[1]  <-  "taxa" 

K <- GAPIT.K(Y=NULL, GD=genotype_data)
heatmap(K$Kinship, symm=TRUE)
```


```{r}
# Using rrBLUP
K <- A.mat(as.matrix(geno))  # additive relationship matrix

any(is.na(K))       
any(is.nan(K))
any(is.infinite(K))
K[is.na(K)] <- 0
K[is.nan(K)] <- 0

pheatmap(K)

pheatmap(K, cluster_rows = TRUE, cluster_cols = TRUE,
         main = "Kinship matrix")
```